src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts

```diff
@@ -256,27 +283,30 @@ export class GeneComparisonToolComponent implements OnInit, AVI, OnDestroy {
         gene.hgnc_symbol.toLowerCase(),
       ];
 
-      if ('Protein - Differential Expression' === this.category) {
+      if (this.category === 'Protein - Differential Expression') {
+        // for this category, the uid will be ensg + uniprotid
         gene.uid += gene.uniprotid;
```

Comment:
Consider pulling this into a new helper method that returns the uid based on the category